app/objects/c_ability.py

```diff
@@ -140,25 +89,23 @@ async def which_plugin(self):
                 return plugin
         return None
 
-    def replace_cleanup(self, encoded_cmd, payload):
-        decoded_cmd = b64decode(encoded_cmd).decode('utf-8', errors='ignore').replace('\n', '')
-        decoded_cmd = decoded_cmd.replace(self.RESERVED['payload'], payload)
-        return decoded_cmd
+    def get_executor(self, platform, name):
+        return next(iter([e for e in self.executors if e.platform == platform and e.name == name]), None)
```

Comment:
This line has some unneeded things happening. If you wanted to do a one liner, you could use a generator expression rather than the list comprehension wrapped within an iterator:

```python
return next((e for e in self.executors if e.platform == platform and e.name == name), None)
```

If you didn't have the `None` default you could drop the inner parens:

```python
return next(e for e in self.executors if e.platform == platform and e.name == name)
```

But really I think it'd just be clearer to write as a loop:

```python
for executor in self.executors:
    if executor.platform == platform and executor.name == name:
        return executor

return None
```

The default value of `None` makes me wonder about a couple things:
1. Should we allow a default value to be passed in rather than return `None`?
2. I often wonder about naming retrieval methods like this `find_foo()` rather than `get_foo()`. IMO, a `find()` method should be allowed to fail (return with no results). Whereas a `get()` method should be assumed to succeed, and therefore throw an error/exception if the retrieval fails. Granted, dictionary `.get()` methods are a wonderful counter-example to that. No real recommendation here.

Reply:
Updated and renamed (good suggestion) `get_executor()` and `get_executors()`